### DISCUSSION TITLE
Complex binding to other controls property

### DISCUSSION THREAD

**Discussion by DrDryg** [AvaloniaUI/Avalonia#12912] [General]
I want to show a canvas based on a dropdown setting. Basically something like this:

`<ContentControl Content="{Binding SelectedPanelControl}"></ContentControl>`

This works fine. When I do a new select in my dropdown (Combobox) the view will set the SelectedPanelControl. But what Im wonder about is if I can use the value of my Combobox to directly set Content using the Value of my Combobox and a dictionary of possible Controls.

So something like this:
```
	<DockPanel>
		<ComboBox Name="comboBox" Height="30" DockPanel.Dock="Top" ItemsSource="{Binding StringData}" SelectedIndex="0">
			<ComboBox.ItemTemplate>
				<DataTemplate x:DataType="x:String">
					<Label Content="{Binding}" FontWeight="Bold"/>
				</DataTemplate>
			</ComboBox.ItemTemplate>
		</ComboBox>
		<Panel>
			<ContentControl Content="{Binding panelControls[key]}"></ContentControl>
		</Panel>		
	</DockPanel>
```

The _key_ in the line
`<ContentControl Content="{Binding panelControls[key]}"></ContentControl>`

Should be taken from comboBox.Item

Is this possible and if so how? Also, "doing the work" in the xaml or the modelview, which is in general preferred?

**Comment by timunie** [comment]
DataTemplates are your friend here: https://github.com/AvaloniaUI/Avalonia.Samples#%EF%B8%8F-datatemplate-samples
